src/core/processing/qgsprocessingutils.cpp

```diff
@@ -761,6 +766,12 @@ QString QgsProcessingUtils::layerToStringIdentifier( const QgsMapLayer *layer )
 
     return QStringLiteral( "%1://%2" ).arg( provider, source );
   }
+
+  if ( layer->dataProvider()->name() == QLatin1String( "annotation" ) && layerName.compare( QLatin1String( "main" ), Qt::CaseInsensitive ) == 0 )
```

Comment:
This could potentially crash on some broken layers (dataProvider() will be nullptr):

```suggestion
  if ( layer->type() == Qgis::LayerType::Annotation && layerName.compare( QLatin1String( "main" ), Qt::CaseInsensitive ) == 0 )
```